### ISSUE TITLE
data:image urls stopped working in layers

### ISSUE THREAD

**Issue by goszczynskip** [uber/deck.gl#3765]
#### Description
Pass inline data url as an iconAtlas to IconLayer (data:image/png;base64,...=). This will return error in console.

This happen after upgrading `deck.gl` from `6.x` to `7.3.2`

#### Environment:
 - Framework Version: [deck.gl 7.3.2]
 - Browser Version: [Firefox 69.0.2]
 - OS: [Windows 10]

#### Logs
```
No valid loader found for data:image/png;base64,<cut out for brevity>
selectLoader select-loader.js:33
    _callee$ parse.js:30
    tryCatch runtime.js:45
    invoke runtime.js:264
    method runtime.js:98
    asyncGeneratorStep asyncToGenerator.js:3
    _next asyncToGenerator.js:25
    _asyncToGenerator asyncToGenerator.js:32
    _asyncToGenerator asyncToGenerator.js:21
    _parse parse.js:49
    parse parse.js:12
    _callee2$ load.js:91
    tryCatch runtime.js:45
    invoke runtime.js:264
    method runtime.js:98
    asyncGeneratorStep asyncToGenerator.js:3
    _next asyncToGenerator.js:25
index.js:1437
    e index.js:1437
    _watchPromise component-state.js:181
```

**Comment by ibgreen** [comment]
Interesting. This is probably an issue for loaders.gl which is now used under the hood to load images. We should add a test case there first. I attached an issue: https://github.com/uber-web/loaders.gl/issues/491

**Comment by kevinresol** [comment]
Also fails for urls without file extension or urls with query parameters

**Comment by crow7m** [comment]
same here

**Comment by Pessimistress** [comment]
Should be fixed in 7.3.6.

**Comment by hijiangtao** [comment]
I met the same problem but with a wired behavior when I want to upgrade my project from deck.gl @v6 to deck.gl @v7. The deck.gl version I used is 7.3.6 already.

I rewrote my customized Layer with new grammar, and passed string to property of `data`, and got thing runs well first, then after I `npm link` it to my demo codes, two fatal errors showed in console:

**Error 1**

> deck: error while initializing invalid
>   Error: Model needs a program
>   ...

**Error 2**

> No valid loader found for 'https://raw.githubusercontent.com/uber-common/deck.gl-data/master/examples/icon/meteorites.json'
> ...

`Model` is a component defined in luma.gl, and data loader is defined in deck.gl. I had no idea why I trigger this situation, even if I modified my code exactly the same as deck.gl's official one.

I finally solved it with reinstalled my project's dependencies, but it seems I didn't figure out the exact reason behind it.

Any ideas on that, or maybe the log can be more concretely? I suspect it's caused by duplicated versions of deck.gl (according to the comments of https://github.com/uber/nebula.gl/issues/114 ), since I'd checked my deck.gl's version to 7.3.6, but why would above errors happened while I didn't modify/update my deps (`node_modules` folder)?
